lib/streamlit/elements/multiselect.py

```diff
@@ -104,6 +103,14 @@ def _check_and_convert_to_indices(
     return [opt.index(value) for value in default_values]
 
 
+def get_default_count(default: Union[Sequence[Any], Any, None]):
+    if default is None:
+        return 0
+    if not isinstance(default, Sequence):
```

Comment:
Checking `isinstance(default, Sequence)` here runs into the same issue where it won't work correctly unless the type of `default` was written with type annotations, which we can't guarantee for all iterable/sequence types (even if it'll almost always be true).

Is it possible to still use `is_iterable` in this check while also keeping the `Sequence[Any]` in the function signature? I'm not sure if the type checker is smart enough to realize that once we get to the final return statement, we must have a `Sequence`, but if it's not, we can just cast to a `Sequence[Any]` before passing the value to the `len` function.

Reply:
Also see my comment in one of the tests for this, because it's possible to run into some real weirdness around the multiselect `default` arg if it's a string.